src/commands/importFromRegistry.ts

```diff
@@ -0,0 +1,168 @@
+        // Enforce the rules from package.json on the server name
+        const name = originalName.toLowerCase().replace(/[^a-z0-9-._~]/g, "~");
+        const getProperty = (property: string) =>
+          vsWinReg.GetStringRegKey(hkeyLocalMachine, path, property);
+
+        if (name !== "" && !config.has("servers." + name)) {
+          newServerNames.push(name);
+          let username: string | undefined;
+          try {
+            username = vsWinReg.GetStringRegKey(
+              "HKEY_CURRENT_USER",
+              "Software\\InterSystems\\Cache\\Servers\\" + originalName,
+              "Server User Name",
+            );
+          } catch (e) {
+            // No-op; will assume the key did not exist (it is valid for it not to).
+          }
+
+          if (username === undefined || username === "") {
+            serversMissingUsernames.push(name);
+          }
+
+          const usesHttps = getProperty("HTTPS") === "1";
+          serverDefinitions[name] = {
+            description: getProperty("Comment"),
+            username: (username === undefined || username === "" ? undefined : username),
+            webServer: {
+              host: getProperty("WebServerAddress") || getProperty("Address"),
+              pathPrefix: getProperty("WebServerInstanceName"),
+              port: parseInt(getProperty("WebServerPort"), 10),
+              scheme: usesHttps ? "https" : "http",
+            },
+          };
+        }
+      });
+
+      if (serversMissingUsernames.length) {
+        let serverName = serversMissingUsernames[0];
+        let username = await vscode.window.showInputBox({
+          ignoreFocusOut: true,
+          placeHolder: "username",
+          prompt: `Username for server '${serverName}'`,
+          validateInput: ((value) => {
+              return value.length > 0 ? "" : "Mandatory field";
+          }),
+        }, cancellationToken);
+        if (cancellationToken.isCancellationRequested || username === undefined) {
+          return;
+        }
+        if (serversMissingUsernames.length > 1) {
+          const items = [
+            `Use username '${username}' for all servers`,
+            `Enter usernames individually for ${serversMissingUsernames.length - 1} more server(s)`,
+            `Cancel`].map((label) => {
+              return { label };
+            });
+          const result = await vscode.window.showQuickPick(items, {
+            canPickMany: false,
+            ignoreFocusOut: true,
+          }, cancellationToken);
+          if (cancellationToken.isCancellationRequested || result === undefined || result.label === items[2].label) {
+            return;
+          } else if (result.label === items[0].label) {
+            for (serverName of serversMissingUsernames) {
+              serverDefinitions[serverName].username = username;
+            }
+          } else {
+            for (serverName of serversMissingUsernames) {
```

Comment:
At the top of this loop I think we need to skip if `serverName===serversMissingUsernames[0]` else won't we get re-prompted for a username for the first no-username server again?